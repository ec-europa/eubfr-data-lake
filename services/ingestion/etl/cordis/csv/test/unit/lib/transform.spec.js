/**
 * @jest-environment node
 */

import mapper from '../../../src/lib/transform';
import testRecord from '../../stubs/record.json';

describe(`DG CORDIS CSV transformer`, () => {
  let result = {};

  beforeAll(() => {
    result = mapper(testRecord);
  });

  test(`Throws an error on empty input`, () => {
    expect(mapper).toThrow();
  });

  test(`Produces correct JSON output structure`, () => {
    expect(result).toMatchSnapshot();
  });

  test(`Project location information is correctly mapped`, () => {
    const firstLocation = result.project_locations[0];
    expect(firstLocation).toMatchSnapshot();
  });
});
