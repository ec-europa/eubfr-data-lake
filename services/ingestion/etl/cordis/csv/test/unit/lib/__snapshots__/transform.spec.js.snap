// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DG CORDIS CSV transformer Produces correct JSON output structure 1`] = `
Object {
  "action": "25",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "EUR",
      "raw": "25",
      "value": 25,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "EUR",
      "raw": "7643751",
      "value": 7643751,
    },
  },
  "call_year": "",
  "description": "SMARTER will develop and deploy innovative strategies to improve Resilience and Efficiency (R&E) related traits in sheep and goats. SMARTER will find these strategies by: i) generating and validating novel R&E related traits at a phenotypic and genetic level ii) improving and developing new genome-based solutions and tools relevant for the data structure and size of small ruminant populations, iii) establishing new breeding and selection strategies for various breeds and environments that consider R&E traits . SMARTER with help from stakeholders chose several key R&E traits including feed efficiency, health (resistance to disease, survival) and welfare. Experimental populations will be used to identify and dissect new predictors of these R&E traits and the trade-off between animal ability to overcome external challenges. SMARTER will estimate the underlying genetic and genomic variability governing these R&E related traits. This variability will be related to performance in different environments including genotype-by-environment interactions (conventional, agro-ecological and organic systems) in commercial populations. The outcome will be accurate genomic predictions for R&E traits in different environments across different breeds and populations. SMARTER will also create a new cooperative European and international initiative that will use genomic selection across countries. This initiative will make selection for R&E traits faster and more efficient. SMARTER will also characterize the phenotype and genome of traditional and underutilized breeds. Finally, SMARTER will propose new breeding strategies that utilise R&E traits and trade-offs and balance economic, social and environmental challenges. The overall impact of the multi-actor SMARTER project will be ready-to-use effective and efficient tools to make small ruminant production resilient through improved profitability and efficiency.",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "H2020",
  "project_id": "772787",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
  ],
  "project_website": "",
  "related_links": Array [
    Object {
      "label": "",
      "url": "",
    },
  ],
  "reporting_organisation": "DEVCO",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "SIGNED",
  "sub_programme_name": "H2020-EU.3.2.1.1.",
  "success_story": "",
  "themes": Array [],
  "third_parties": Array [
    Object {
      "address": "",
      "country": "/RIA",
      "email": "",
      "name": "H2020-SFS-2017-2",
      "phone": "",
      "region": "",
      "role": "coordinator",
      "type": "",
      "website": "",
    },
  ],
  "timeframe": Object {
    "from": "2018-11-01",
    "from_precision": "year",
    "to": "2022-10-31",
    "to_precision": "year",
  },
  "title": "SMAll RuminanTs breeding for Efficiency and Resilience",
  "type": Array [],
}
`;

exports[`DG CORDIS CSV transformer Project location information is correctly mapped 1`] = `
Object {
  "address": "",
  "centroid": null,
  "country_code": "",
  "location": null,
  "nuts": Array [],
  "postal_code": "",
  "region": "",
  "town": "",
}
`;
